react/components/Modal/useModal.ts

```diff
@@ -0,0 +1,10 @@
+import { useState } from 'react'
+
+const useModal = () => {
```

Comment:
I think we could rename this hook to something like `useDisclosure` just like [ChackraUI](https://chakra-ui.com/usedisclosure) does.

I also think that it should be a separate component and not a part of the modal. Honestly I don't like none `useDisclosure`, we could see other suggestions.

Reply:
I agree we can rename it. I created this hook because I noticed that everyone uses the Modal in the same way that the hook is defined